app/models/topic.rb

```diff
@@ -126,6 +126,11 @@ def assign(user_id=2, assigned_to)
     self.save
   end
 
+  def self.bulk_assign(post_attributes, assigned_to)
+    Post.create(post_attributes)
+    self.update_all(assigned_user_id: assigned_to, current_status: 'pending')
```

Comment:
This seems like a (big) problem; are you sure that `self.update_all` will not update _all_ topics (not just those which were created)?

I think a it's necessary to do:

```ruby
new_posts = Post.create(post_attributes)
new_posts.update_all(assigned_user_id: assigned_to, current_status: 'pending')
```

... am I correct?